File: backend/core/bases.py
# Canonical MARS bases — keep in sync with `src/app/data/bases.ts`.

ORGANIZATION_BASES = (
    'Harare',
    'Bulawayo',
    'Gweru',
    'Masvingo',
    'Mutare',
    'Victoria Falls',
    'Marondera',
)

ORGANIZATION_BASES_SET = frozenset(ORGANIZATION_BASES)

DEFAULT_BASE = ORGANIZATION_BASES[0]


def normalize_base(raw: str) -> str | None:
    s = (raw or '').strip()
    if not s:
        return None
    if s in ORGANIZATION_BASES_SET:
        return s
    # common misspelling from older forms
    if s.lower() == 'bulwayo':
        return 'Bulawayo'
    return None


def resolve_base(raw: str) -> str:
    """Return a valid base; default to Harare if empty or unknown."""
    n = normalize_base(raw)
    return n if n else DEFAULT_BASE

File: src/app/data/bases.ts
/**
 * Canonical MARS operational bases — keep in sync with `backend/core/bases.py`.
 * (Spelling: Bulawayo — standard city name.)
 */
export const ORGANIZATION_BASES = [
  'Harare',
  'Bulawayo',
  'Gweru',
  'Masvingo',
  'Mutare',
  'Victoria Falls',
  'Marondera',
] as const;

export type OrganizationBase = (typeof ORGANIZATION_BASES)[number];

export const DEFAULT_BASE: OrganizationBase = 'Harare';

export function resolveBase(raw: string | undefined): OrganizationBase {
  const s = (raw || '').trim();
  if ((ORGANIZATION_BASES as readonly string[]).includes(s)) return s as OrganizationBase;
  if (s.toLowerCase() === 'bulwayo') return 'Bulawayo';
  return DEFAULT_BASE;
}
